### ISSUE TITLE
Joining Models that isn't linked in the Data Model causes no metadata inferrence

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#27323]
**Describe the bug**
Working with Models that are not linked in the Data Model causes no metadata to be inferred, which makes it impossible to use certain fields and their display name will reference the ID, making it very confusing.

Workaround: Create a View directly on the database instead of a Model.

**To Reproduce**
1. New GUI Question > Sample Database > Products - save as "M1", turn into a model
2. New GUI Question > Sample Database > People - save as "M2", turn into a model
3. Go to M1, join M2 on ID = ID
4. Add filter by Created At on M2 (e.g. created between 01/01/2000 and now)
5. Summarize by distinct ID on M2, group by Source on M2
6. Visualize it as table
![image](https://user-images.githubusercontent.com/1447303/208452215-3c882414-065d-4828-93b8-96acd402bf7c.png)

Column names are a mixed bag of confusion
![image](https://user-images.githubusercontent.com/1447303/208453085-b89ea9b2-be47-4161-a23d-ddb3b23603fd.png)

**Information about your Metabase Installation:**
Tested 0.45.1

**Additional context**
Related to #26849 and #21646/#12930

**Comment by dpsutton** [comment]
Waiting to hear more concrete issues. I don't see anything about metadata in this issue. Seems to be more a feature request to use the model's name rather than "Question 1234" in aliases and display names.

**Comment by dpsutton** [comment]
I think this needs to be fixed with a small change to the BE, and then a change to either the query builder or metabase-lib.

The display names are driven off of the alias chosen in the query:

```clojure
aliases=> (->> (qp/process-query {:database 1,
                                  :query {:source-table "card__1302",
                                          :joins [{:fields :all,
                                                   :source-table "card__1303",
                                                   :condition [:=
                                                               [:field 60 nil]
                                                               [:field
                                                                46
                                                                {:join-alias "Question 1303"}]],
                                                   :alias "Question 1303"}],
                                          :limit 2},
                                  :type :query})
               :data :results_metadata :columns (map :display_name)
               )
("ID"
 "Title"
 "Category"
 "Created At"
 "Question 1303 → ID"
 "Question 1303 → Name"
 "Question 1303 → Source"
 "Question 1303 → Created At")
```
If the aliases are instead changed to "m2" we get the following display names on the resulting query:
`("ID" "Title" "Category" "Created At" "m2 → ID" "m2 → Name" "m2 → Source" "m2 → Created At")`

The alias is chosen by the FE when doing the join. When you choose a model to join against, the FE issues a query like `http://localhost:3000/api/table/card__1303/query_metadata` to get the metadata. This comes back _without_ information that what you have chosen is in fact a model.
```javascript
{
  // no indication that this is a model
  "id": "card__1303",
  "db_id": 1,
  "display_name": "m2",
  "schema": "27323",
  "moderated_status": null,
  "description": null,
  "fields": [
    { "display_name": "ID", ... },
    { "name": "NAME", ... },
    { "name": "SOURCE", ... },
    { "name": "CREATED_AT", ... }
  ],
  "dimension_options": {...}
}
```

If the backend includes the information that this is a model, then the FE can adapt it's name as the alias rather than turning `"id": "card__1303"` into the alias of `"Question 1303"`.

**Comment by kamilmielnik** [comment]
Reproducible in 0.45.1 ([screenshot](https://github.com/metabase/metabase/assets/6830683/073efb71-e123-4d67-a8ad-5c7441d8dc09)).
Fixed in `master` ([d35c779](https://github.com/metabase/metabase/commit/d35c779976ae6bd79369e9cc6bec8540908f37d2)) - [screenshot](https://github.com/metabase/metabase/assets/6830683/f7b5d328-eaed-4bee-a191-a5a5f1a9034b).
Marking as fixed in the next version (48).
Closing.

**Comment by snoe** [comment]
Reopening to add test.
